lib/Migration/Version11000Date20201209142525.php

```diff
@@ -0,0 +1,144 @@
+
+namespace OCA\Talk\Migration;
+
+use Closure;
+use Doctrine\DBAL\Types\Types;
+use OCP\DB\ISchemaWrapper;
+use OCP\IDBConnection;
+use OCP\Migration\IOutput;
+use OCP\Migration\SimpleMigrationStep;
+
+class Version11000Date20201209142525 extends SimpleMigrationStep {
+	/** @var IDBConnection */
+	protected $connection;
+
+	public function __construct(IDBConnection $connection) {
+		$this->connection = $connection;
+	}
+
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		$changedSchema = false;
+		if (!$schema->hasTable('talk_internalsignaling')) {
+			$table = $schema->createTable('talk_internalsignaling');
+
+			// Auto increment id
+			$table->addColumn('id', Types::BIGINT, [
+				'autoincrement' => true,
+				'notnull' => true,
+			]);
+
+			$table->addColumn('sender', Types::STRING, [
+				'notnull' => true,
+				'length' => 255,
+			]);
+			$table->addColumn('recipient', Types::STRING, [
+				'notnull' => true,
+				'length' => 255,
+			]);
+			$table->addColumn('message', Types::TEXT, [
+				'notnull' => true,
+			]);
+			$table->addColumn('timestamp', Types::INTEGER, [
+				'notnull' => true,
+				'length' => 11,
+			]);
+
+			$table->setPrimaryKey(['id']);
+			$table->addIndex(['recipient', 'timestamp'], 'tis_recipient_time');
+
+			$changedSchema = true;
+		}
+
+		if (!$schema->hasTable('talk_guestnames')) {
+			$table = $schema->createTable('talk_guestnames');
+
+			// Auto increment id
+			$table->addColumn('id', Types::BIGINT, [
+				'autoincrement' => true,
+				'notnull' => true,
+			]);
+
+			$table->addColumn('session_hash', Types::STRING, [
+				'notnull' => false,
+				'length' => 64,
+			]);
+			$table->addColumn('display_name', Types::STRING, [
+				'notnull' => false,
+				'length' => 64,
+				'default' => '',
+			]);
+
+			$table->setPrimaryKey(['id']);
+			$table->addUniqueIndex(['session_hash'], 'tg_session_hash');
+			$changedSchema = true;
+		}
+
+		return $changedSchema ? $schema : null;
+	}
+
+	/**
+	 * @param IOutput $output
+	 * @param Closure $schemaClosure The `\Closure` returns a `ISchemaWrapper`
+	 * @param array $options
+	 */
+	public function postSchemaChange(IOutput $output, Closure $schemaClosure, array $options): void {
+		if (!$this->connection->tableExists('talk_guests')) {
+			return;
+		}
+
+		$insert = $this->connection->getQueryBuilder();
+		$insert->insert('talk_guestnames')
+			->values([
+				'session_hash' => $insert->createParameter('session_hash'),
+				'display_name' => $insert->createParameter('display_name'),
+			]);
+
+		$query = $this->connection->getQueryBuilder();
+		$query->select('*')
+			->from('talk_guests');
+
+		$result = $query->execute();
+		while ($row = $result->fetch()) {
```

Comment:
I guess there was no way to make this into a single query like "insert into", which might not work with all DBs.

hope the number of guests is low on most setup so that the downtime is minimal

Reply:
We did worse copies on other updates. But well that's why we try not to backport such things